hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/router/TestRouterRpc.java

```diff
@@ -2070,6 +2071,23 @@ public void testSetBalancerBandwidth() throws Exception {
     }, 100, 60 * 1000);
   }
 
+  @Test
+  public void testRecordRealClientIp() {
+    GenericTestUtils.LogCapturer logger =
+        GenericTestUtils.LogCapturer.captureLogs(Server.LOG);
+    CallerContext.setCurrent(
+        new CallerContext.Builder(
+            "clientContext,clientIp:2.2.2.2,clientPort:1234").build());
+    try {
+      String filePath = "/test/f.log";
+      routerProtocol.getBlockLocations(filePath, 0, 100);
+    } catch (Exception e) {
+      // do nothing
+    } finally {
+      assertTrue(logger.getOutput().contains("2.2.2.2:1234"));
```

Comment:
Can we test with HADOOP_CALLER_CONTEXT_ENABLED_KEY set to true and false?

Reply:
Thanks @goiri   for the comment and review.
I will update later.